amd_openvx_extensions/amd_rpp/source/tensor/Resample.cpp

```diff
@@ -0,0 +1,251 @@
+/*
+Copyright (c) 2019 - 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "internal_publishKernels.h"
+#include "vx_ext_amd.h"
+
+struct ResampleLocalData 
+{
+    vxRppHandle *handle;
+    Rpp32u deviceType;
+    RppPtr_t pSrc;
+    RppPtr_t pDst;
+    float quality;
+    RpptDescPtr pSrcDesc;
+    RpptDescPtr pDstDesc;
+    Rpp32s *sampleFrames;
+    Rpp32s *sampleChannels;
+    Rpp32f *inRateTensor;
+    Rpp32f *outRateTensor;
+    size_t inputTensorDims[RPP_MAX_TENSOR_DIMS];
+    size_t outputTensorDims[RPP_MAX_TENSOR_DIMS];
+};
+
+void update_destination_roi(const vx_reference *parameters, ResampleLocalData *data, RpptROI* src_roi, RpptROI* dst_roi) {
+    float _scale_ratio;
+    for(unsigned i = 0; i < data->pSrcDesc->n; i++) {
+        if (data->inRateTensor[i] != 0)
+            _scale_ratio = data->outRateTensor[i] / (float)data->inRateTensor[i];
+        else
+            _scale_ratio = 0;
+        dst_roi[i].xywhROI.xy.x = (int)std::ceil(_scale_ratio * src_roi[i].xywhROI.xy.x);
+        dst_roi[i].xywhROI.xy.y = src_roi[i].xywhROI.xy.y;
+    }
+}
+
+static vx_status VX_CALLBACK refreshResample(vx_node node, const vx_reference *parameters, vx_uint32 num, ResampleLocalData *data) {
+    vx_status status = VX_SUCCESS;
+    STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[5], 0, data->pSrcDesc->n, sizeof(float), data->inRateTensor, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+    STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[4], VX_TENSOR_BUFFER_HOST, &data->outRateTensor, sizeof(data->outRateTensor)));
+    void *roi_tensor_ptr_src, *roi_tensor_ptr_dst;
+    if (data->deviceType == AGO_TARGET_AFFINITY_GPU)
+    {
+#if ENABLE_OPENCL
+        return VX_ERROR_NOT_IMPLEMENTED;
+#elif ENABLE_HIP
+        return VX_ERROR_NOT_IMPLEMENTED;
```

Comment:
Would suggest not to return VX_ERROR_NOT_IMPLEMENTED for HIP backend for refresh alone
since there is no dependency with RPP implementation for this buffer query